File: backend/app/models/product.py
from pydantic import BaseModel
from typing import List, Optional


class Product(BaseModel):
    """Product model for kiosk items"""
    pid: str  # product_id
    name: str
    price: float
    description: str = ""
    image_url: str = ""
    tags: List[str] = []
    available: bool = True


class CreateProductRequest(BaseModel):
    """Request model for creating a new product"""
    name: str
    price: float
    description: str = ""
    image_url: str = ""
    tags: List[str] = []


class CreateProductResponse(BaseModel):
    """Response model for product creation"""
    pid: str  # product_id


class UpdateProductRequest(BaseModel):
    """Request model for updating a product"""
    name: Optional[str] = None
    price: Optional[float] = None
    description: Optional[str] = None
    image_url: Optional[str] = None
    tags: Optional[List[str]] = None
    available: Optional[bool] = None


class ProductListResponse(BaseModel):
    """Response model for product list"""
    products: List[Product]
    total: int

File: frontend-kiosk/src/api/product.js
import { request } from "./client.js";

/**
 * 제품 정보 타입
 * @typedef {Object} Product
 * @property {string} pid - 제품 ID
 * @property {string} name - 제품 이름
 * @property {number} price - 제품 가격 (원/g)
 * @property {string} description - 제품 설명
 * @property {string} image_url - 제품 이미지 URL
 * @property {string[]} tags - 제품 태그
 * @property {number} original_price - 원래 가격
 * @property {number} original_gram - 원래 무게
 */

/**
 * 전체 제품 목록 조회
 * @returns {Promise<Product[]>} 제품 목록
 */
export async function getProducts() {
  return request("/products/");
}

/**
 * 특정 제품 상세 정보 조회
 * @param {string} productId - 제품 ID
 * @returns {Promise<Product>} 제품 정보
 */
export async function getProductById(productId) {
  return request(`/products/${productId}`);
}
